includes/class-eddlicense.php

```diff
@@ -0,0 +1,377 @@
+<?php
+/**
+ * Gift LearnDash Courses.
+ *
+ * @package   learndash-gift-edd
+ * @author    MissionLab <missionlab.dev>
+ * @license   GPL-2.0+
+ * @link      https://missionlab.dev
+ *
+ */
+
+/**
+ * Class that handles licensing for EDD
+ *
+ * @package learndash-gift-edd/include/
+ * @author  MissionLab
+ */
+class GiftEddLicense {
+
+	/**
+	 * Instance of this class.
+	 *
+	 * @since    1.0.0
+	 *
+	 * @var      object
+	 */
+	protected static $instance = null;
+
+	protected static $_prefix = 'learndash-gift-edd';
+
+	/**
+	 * Initialize the plugin by setting localization and loading public scripts
+	 * and styles.
+	 *
+	 * @since     1.0.0
+	 */
+	public function __construct() {
+		add_action( 'admin_init', array( $this, 'activate_license' ), 100 );
+		add_action( 'admin_init', array( $this, 'deactivate_license' ), 100 );
+		add_action( 'admin_init', array( $this, 'check_license' ) );
+		add_action( 'admin_init', array( $this, 'register_settings' ), 50 );
+		add_action( 'admin_init', array( $this, 'plugin_updater' ) );
+		add_action( 'admin_notices', array( $this, 'license_admin_notice' ) );
+	}
+
+
+	/**
+	 * Return an instance of this class.
+	 *
+	 * @return    object    A single instance of this class.
+	 * @since     1.0.0
+	 *
+	 */
+	public static function get_instance() {
+
+        // If the single instance hasn't been set, set it now.
+		if ( null == self::$instance ) {
+			self::$instance = new self;
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Return the plugin prefix
+	 *
+	 * @return string
+	 *
+	 */
+
+	public static function get_prefix() {
+		return self::$_prefix;
+	}
+
+	/**
+	 * Initialize plugin license key option
+	 */
+
+	public function register_settings() {
+		register_setting( 'learndash_edd_gift_options', self::$_prefix . '_license_key', array( $this, 'sanitize_license' ) );
+    }
+
+	/**
+	 * Check if license key needs to be reactivate
+	 */
+
+	public function sanitize_license( $new ) {
+		$old = get_option( self::$_prefix . '_license_key' );
+		if ( $old && $old != $new ) {
+			delete_option( self::$_prefix . '_license_key' ); // new license has been entered, so must reactivate
+		}
+
+		return $new;
+	}
+
+	/**
+	 * Return the license key
+	 *
+	 * @return string
+	 * @author Tanner Moushey
+	 */
+	public static function get_license_key() {
+		return get_option( self::$_prefix . '_license_key' );
+	}
+
+	/**
+	 * Return the license status
+	 *
+	 * @return string
+	 * @author Tanner Moushey
+	 */
+	public static function get_license_status() {
+		return trim( get_option( self::$_prefix . '_license_status' ) );
+	}
+
+	/**
+	 * Handle License activation
+	 */
+	public function activate_license() {
+
+		// listen for our activate button to be clicked
+		if ( ! isset( $_POST[ self::$_prefix . '_license_activate' ]) ) {
```

Comment:
This may just be Github, but it seems that line indentation is really wonky.

Reply:
Yep, this is just GH
<img width="221" alt="Screen Shot 2021-02-25 at 2 16 48 PM" src="https://user-images.githubusercontent.com/20672471/109236652-25432080-7774-11eb-9b3c-dc770fa53d90.png">

Reply:
Fixed